WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueSocialFragment.java

```diff
@@ -0,0 +1,364 @@
+        outState.putString(KEY_USERNAME, mUsername);
+    }
+
+    @Override
+    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+    }
+
+    @Override
+    public void onTextChanged(CharSequence s, int start, int before, int count) {
+    }
+
+    @Override
+    public void afterTextChanged(Editable s) {
+    }
+
+    @Override
+    protected void onHelp() {
+    }
+
+    @Override
+    protected void onLoginFinished() {
+        endProgress();
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onAccountChanged(OnAccountChanged event) {
+        if (event.isError()) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UPDATE_DISPLAY_NAME_FAILED);
+            AppLog.e(AppLog.T.API, "SignupEpilogueSocialFragment.onAccountChanged: " +
+                    event.error.type + " - " + event.error.message);
+            endProgress();
+            showErrorDialog(getString(R.string.signup_epilogue_error_generic));
+        } else if (changedUsername()) {
+            startProgress();
+            PushUsernamePayload payload = new PushUsernamePayload(mUsername,
+                    AccountUsernameActionType.KEEP_OLD_SITE_AND_ADDRESS);
+            mDispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload));
+        } else if (event.causeOfChange == AccountAction.PUSH_SETTINGS && mSignupEpilogueListener != null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UPDATE_DISPLAY_NAME_SUCCEEDED);
+            mSignupEpilogueListener.onContinue();
+        }
+    }
+
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onUsernameChanged(OnUsernameChanged event) {
+        if (event.isError()) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UPDATE_USERNAME_FAILED);
+            AppLog.e(AppLog.T.API, "SignupEpilogueSocialFragment.onUsernameChanged: " +
+                    event.error.type + " - " + event.error.message);
+            endProgress();
+            showErrorDialog(getString(R.string.signup_epilogue_error_generic));
+        } else if (mSignupEpilogueListener != null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SIGNUP_SOCIAL_EPILOGUE_UPDATE_USERNAME_SUCCEEDED);
+            mSignupEpilogueListener.onContinue();
+        }
+    }
+
+    protected boolean changedDisplayName() {
+        return !StringUtils.equals(getArguments().getString(ARG_DISPLAY_NAME), mDisplayName);
+    }
+
+    protected boolean changedUsername() {
+        return !StringUtils.equals(getArguments().getString(ARG_USERNAME), mUsername);
+    }
+
+    protected void showErrorDialog(String message) {
+        DialogInterface.OnClickListener dialogListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                switch (which) {
```

Comment:
That seems subjective since it doesn't make the `switch` statement more readable.  If the case is excluded, it is ignored.  If the case is added with only a comment, it is ignored.  It's effectively the same.  Looking at the statement now, it can be seen which cases produce actions simplifying readability.  Adding a case just for a comment seems too verbose.

Reply:
Cool, let's just add the comment, quickly explaining that the neutral case is ignored as the dialog is being let to simply dismiss. This makes it clear that it's a deliberate functionality. Thanks!

Reply:
I added a comment in 41a48a7ddb1de6804276264c0e84c19f8d81c4f7.